### ISSUE TITLE
Refresh Rate not set to 165

### ISSUE THREAD

**Issue by getBoolean** [RickStrahl/SetResolution#8]
I've created these two profiles below. When I switch to `ipad`, the refresh rate is switched to `60`, but when I switch to `default`, the refresh rate stays at `60` instead of switching to `165`. Any idea what I am doing wrong?

```
default:  3440 x 1440, 32, 165
ipad:  2882 x 2160, 32, 60
```

Device Details:
- Windows 11
- AMD Radeon RX 7900 XT GPU

**Comment by HopleViorn** [comment]
The same as you are. My profile is 3440x1440,144 and 2880x1920,120. Every time I switch to this 3440x1440 it will be 120hz but not 144hz.

**Comment by RickStrahl** [comment]
Can you switch to these resolutions and refresh rates with the UI?

What about explicitly running from the command line?

**Comment by getBoolean** [comment]
@RickStrahl do you mean from Windows settings? yes, I can change resolution and refresh rate to these values manually

using the `sr` command-line tool? see issue description

I'm no longer using SR, I found an alternative solution for my use case that doesn't require switching resolution.

**Comment by RickStrahl** [comment]
Yes - I was asking because the original issue talks about using a profile configuration (which should work the same, but better to make sure). I was looking for the actual message that is returned if there is one.

I'm not sure why it wouldn't work with certain refresh rates. On my machine I'm able to set to any of the supported refresh rates, but my rates are very low (ie. standard 60hz). Whatever resolution is chose has to match one of the listed modes **exactly**.

To list all available modes (including orientations if non-default):

```ps
setresolution list -la
```

**Comment by Unix0815** [comment]
It's even worse in my case. I want to switch to 720x576@50 Hz and your tool switches to 1920x1080@120 Hz instead. That's not even close ;)

I think it has some problems with screens that can display at multiple different refresh rates. TV's for example

**Comment by TopCheddar27** [comment]
Just to add to this, the -f switch is broken. For reference, when using this on a 240hz screen to set it to 60hz the command completes successfully with correct output:

![image](https://github.com/RickStrahl/SetResolution/assets/73973868/96df0005-0752-470e-965c-ec83a0453e11)

However it clearly does not switch the framerate:

![image](https://github.com/RickStrahl/SetResolution/assets/73973868/33487d07-9073-4b63-b666-c8ebae4a1e22)

**Comment by KyleMizell** [comment]
I'm also having trouble changing the refresh rate. I can change it manually through the Windows 11 UI, but not through your script from the command line. It changes the resolution just fine.

![image](https://github.com/RickStrahl/SetResolution/assets/1577615/4bb74d01-511d-4069-9734-c2e40eb396ff)
![image](https://github.com/RickStrahl/SetResolution/assets/1577615/6fc5ce98-a802-407f-a61e-959d4d7fcd02)

It reports that it was able to change the refresh rate, but it stays at 60hz.

**Comment by khalil-shao** [comment]
It looks like `-f` gets overridden by the default refresh rate from `SetResolution.xml` when using a predefined profile like `4K` or `1080p`.

As a workaround, you can use `-w` and `-h` to specify the resolution directly, which avoids the issue. You can also update `SetResolution.xml` with the refresh rate you want.

The bug appears to be in `SetResolutionProcessor.cs` around line 50. I’ll try to put together a fix when I get a chance.

**Comment by tofu-dreg** [comment]
I think I'm seeing this issue. I upgraded from a 1080p to 1440p monitor and now when switching to higher Nvidia DSR resolutions, the refresh rate will switch from 240 to 120 as it is the max supported by these higher resolutions, but upon returning to native via "set -w 2560 -h 1440 -f 240", the refresh rate stays at 120 and does not return to 240, needing to be manually changed in NVCP or windows settings.

**Comment by segevfiner** [comment]
I just tried: `.\SetResolution set -w 2560 -h 1440 -f 165 -nopersist` and it set my monitor to 60hz instead of 165 as requested...
